domain/src/main/java/org/oppia/android/domain/exploration/lightweightcheckpointing/ExplorationCheckpointController.kt

```diff
@@ -228,6 +236,34 @@ class ExplorationCheckpointController @Inject constructor(
     }
   }
 
+  fun isSavedCheckpointCompatibleWithExploration(
+    profileId: ProfileId,
+    explorationId: String
+  ): DataProvider<Boolean> {
+    return retrieveCacheStore(profileId)
+      .transformAsync(
+        CHECK_IS_EXPLORATION_CHECKPOINT_COMPATIBLE_WITH_EXPLORATION_DATA_PROVIDER_ID
+      ) { explorationCheckpointDatabase ->
+        val checkpoint = explorationCheckpointDatabase.explorationCheckpointMap[explorationId]
+        val chapterDataJsonObject = jsonAssetRetriever.loadJsonFromAsset("$explorationId.json")
+
+        if (checkpoint != null) {
+          val t = chapterDataJsonObject?.getInt("version")
+          val g = checkpoint.explorationVersion
+          Log.d("12345", "isSavedCheckpointCompatibleWithExploration: $g $t")
+          AsyncResult.success(
+            chapterDataJsonObject?.getInt("version") == checkpoint.explorationVersion
+          )
+        } else
+          AsyncResult.failed(
+            ExplorationCheckpointNotFoundException(
+              "Checkpoint with the explorationId $explorationId was not found " +
+                "for profileId ${profileId.internalId}."
+            )
+          )
+      }
+  }
+
```

Comment:
This code is responsible for matching the exploration version with the version for which the checkpoint was saved.